src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -249,16 +250,32 @@ private function formatListForPresentation(CombinationListForEditing $combinatio
                 'isSelected' => false,
                 'name' => $combination->getCombinationName(),
                 'reference' => $combination->getReference(),
-                //@todo: don't forget image path when implemented in the query
                 'impactOnPrice' => (string) $combination->getImpactOnPrice(),
                 'quantity' => $combination->getQuantity(),
                 'isDefault' => $combination->isDefault(),
+                'imageUrl' => $this->formatImageUrl($combination->getImagePath()),
             ];
         }
 
         return $data;
     }
 
+    /**
+     * @param string|null $imgPath
+     *
+     * @return string
+     */
+    private function formatImageUrl(?string $imgPath): string
+    {
+        if (!$imgPath) {
+            $imagePathFactory = $this->get('prestashop.adapter.product.image.product_image_url_factory');
+
+            return $imagePathFactory->getNoImagePath(ProductImagePathFactory::IMAGE_TYPE_SMALL_DEFAULT);
+        }
+
+        return $this->getContext()->link->getAdminBaseLink() . ltrim($imgPath, '/');
```

Comment:
This shouldn't be required any more, since it's already handled by `prestashop.adapter.product.image.product_image_url_factory`

Reply:
Also since it's always the same url for no image, I think you should get it only once before the loop in `formatListForPresentation` And use `'imageUrl' => $combination->getImagePath() ?: $noImageUrl,`

Reply:
it is still required :sweat:
Query handler returns relative path if image exists, then I need to prepend the "domain based url" here in controller. (that was the idea - to avoid manipulating url inside query handler, but use it here in controller instead).

So another way would be returning already formatted url from a query ( but i think the url thing is the responsibility of controller)

Reply:
what about intermediate solution - a new method `buildFullPath(string $relativePath): string` in a factory?
https://github.com/PrestaShop/PrestaShop/pull/24013/files#diff-67c4e1ce3c380fb648c732b0f080b8719848ec9d697ff5539fb5a79c5fe77dc6R93